rest_framework/authentication.py

```diff
@@ -19,7 +20,7 @@ def get_authorization_header(request):
     Hide some test client ickyness where the header can be unicode.
     """
     auth = request.META.get('HTTP_AUTHORIZATION', b'')
-    if isinstance(auth, type('')):
```

Comment:
I believe we've just changed the meaning here. `string_types` includes both binary and unicode. `type('')` with `from __future__ import unicode_literals` will always only be unicode.

I think we should probably have `if isinstance(auth, text_type)`, to preserve the meaning, as the intent is to encode unicode into bytestrings.